### PR TITLE
feat: update the blog page

### LINKED ISSUE #470: support tag and count for our blog
## Actual Behavior

At the moment, our blog only shows the title, author and other information in the summary page, 

for example:

image

## Expected Behavior

We need to add some tags for each blog, and at the same time, viewers can search blogs via tags, so that it will be easier for viewer to get what they want

for example:

https://quickwit.io/blog

image

https://developer.axioms.io/blog

image

### PR DESCRIPTION
**Pull request by guoqqqi**
Changes:
image
image

1. Left sidebar replaced with tags

Screenshots of the change:

resolve https://github.com/apache/apisix-website/issues/470

### CODE REVIEW

**Comment by guoqqqi** [comment]
image
I am not currently adding too much styling to the header page and I have a few problems that I need help with.
1. if I add a background colour or image to the header section, it doesn't seem to match the style of the other pages, like the team etc page
image

2. there seems to be some empty space and I'm torn between removing it and adding the header, but I think it looks good too, it just looks like something is missing

cc @juzhiyuan @1502shivam-singh @Baoyuantop @bzp2010 @liuxiran

File: website/src/theme/BlogPostItem/index.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
```
Comment (juzhiyuan) [inline website/src/theme/BlogPostItem/index.js:4]: This file is copied from Facebook?

Comment (guoqqqi) [reply website/src/theme/BlogPostItem/index.js:4]: The file is a commit of the PR I recovered under https://github.com/apache/apisix-website/pull/616 and it appears to be a shell CLI generated file, https://docusaurus.io/docs/cli#docusaurus-swizzle-sitedir
cc @Baoyuantop have a look

File: website/src/theme/BlogTagsPostsPage/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import Layout from '@theme/Layout';
+import BlogPostItem from '@theme/BlogPostItem';
+import BlogSidebar from '@theme/BlogSidebar';
+import { ThemeClassNames } from '@docusaurus/theme-common'; // Very simple pluralization: probably good enough for now
+
+function BlogTagsPostPage(props) {
+  const { metadata, items } = props;
+  const { name: tagName } = metadata;
+  const [, setTagsCount] = useState();
+  const tagsTotal = JSON.parse(localStorage.getItem('tagsTotal'));
```
Comment (1502shivam-singh) [inline website/src/theme/BlogTagsPostsPage/index.js]: Will pushing the definition of `tagsTotal` in useEffect cause some errors or unwanted behaviour ?

File: website/src/theme/BlogPostItem/index.js (hunk 1 above)
Comment (1502shivam-singh) [reply website/src/theme/BlogPostItem/index.js:4]: Yup, the blog page is not so customisable directly, so docusaurus has this functionality to `swizzle` out elements,
Refer to this - [`swizzle`](https://github.com/apache/apisix-website/issues/490#issuecomment-911850832)

**Comment by 1502shivam-singh** [comment]
> 2. there seems to be some empty space and I'm torn between removing it and adding the header, but I think it looks good too, it just looks like something is missing

Absence of balancing visual element i guess (check example below). Though, it seems trivial to me for now, we can work on it later maybe.

image

the dots graphic and slight color change does that here. We can too put something less focus wanting, but a balancing visual element in the right or left area for some balance.

Though, this one looks fine too.

**Review by juzhiyuan — APPROVED** [review]

**Review by bzp2010 — APPROVED** [review]

**Pull request merged**